docs/adrs/00005-license-export-analytics.md

```diff
@@ -0,0 +1,114 @@
+
+## Status
+
+DRAFT
+
+## Context
+### An SBOM typically contains three types of license information:
+
+#### 1 License information for the entire SBOM
+
+
+##### 1.1 CycloneDX SBOM-level License
+
+```json
+"licenses" : [
+  { 
+    "license" : { 
+      "id" : "Apache-2.0" 
+    }
+  }
+]
+```
+
+##### 1.2 SPDX SBOM-level License
+```json
+"dataLicense": "CC0-1.0",
+```
+If I understand correctly, CycloneDX's SBOM-level License refers to the license of the package corresponding to the SBOM, while SPDX's SBOM-level License refers to the license of the SBOM document itself.
+#### 2 Package-level licenses
+##### 2.1 CycloneDX
+Contains three types: Id, name, and Expression
+The Expression follows the SPDX expression standard.
+```json
+"licenses" : [  {    "license" : {      "id" : "EPL-1.0"    }  },  {    "license" : {      "name" : "GNU Lesser General Public License",      "url" : "http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html"    }  }],`
+
+```
+##### 2.2 SPDX
+It's Expression and follows the SPDX expression standard.
+```json
+"licenseDeclared": "Apache-2.0 OR LicenseRef-GPL-2.0-with-classpath-exception",
+```
+##### 2.3 Special Types in SPDX: ExtractedLicensingInfos
+```json
+"hasExtractedLicensingInfos": [
+{
+"comment": "External License Info is obtained from a build system which predates the SPDX specification and is not strict in accepting valid SPDX licenses.",
+"extractedText": "The license info found in the package meta data is: GPL-2.0-with-classpath-exception. See the specific package info in this SPDX document or the package itself for more details.",
+"licenseId": "LicenseRef-GPL-2.0-with-classpath-exception",
+"name": "GPL-2.0-with-classpath-exception"
+},
+```
+The licenseId can be associated with the license in the package.
+### 2 Current license export requirements
+https://docs.google.com/spreadsheets/d/1Ki9hkLl94L3G4A4rfQQ4wuEnQFTPkpNflClVjlk2mQA/edit?gid=0#gid=0
+
+### 3 Current license data structure design in trustify
+
+```rust
+pub struct Model {
+    #[sea_orm(primary_key)]
+    pub id: Uuid,
+    pub text: String,
+    pub spdx_licenses: Option&lt;Vec&lt;String&gt;&gt;,
+    pub spdx_license_exceptions: Option&lt;Vec&lt;String&gt;&gt;,
+}
+```
+The text field corresponds to CycloneDX's Id, Name, or complete Expression. When it's an Expression, it will be parsed into a license array and stored in spdx_licenses, while parsing exceptions are stored in spdx_license_exceptions.
+For SPDX, the text field corresponds to the complete Expression. Similarly, the Expression will be parsed into a license array and stored in spdx_licenses, with parsing exceptions stored in spdx_license_exceptions.
+Currently missing SPDX's ExtractedLicensingInfos and SBOM-level licenses.
+
+### 4 License data structure in GUAC
+![erd_guac.png](erd_guac.png)
+GUAC parses all expressions into individual licenses and saves them separately. LicenseRef* represents ExtractedLicensingInfos, with its extractedText stored in the inline field.
```

Comment:
This feels wrong. As a SPDX license expression, to my understanding, is not simply a "list of licenses", but actually, indeed an expression. Which might say "either, or". Or even more complex expressions.

Reply:
yes. SPDX license expression includes not only individual license information but also the relationships between these licenses. It's like the shortcomings of this design that I mentioned below.However, there must be a single license name to associate with the license ID in ExtractedLicensingInfos.

Reply:
If we retain the current license structure and add a new entity to store the individual license name information, how would that work?

Reply:
Storing the license, I do understand that you need to store individual licenses. However, this is more about the license a project declares. IIRC, we already have a full list of SPDX licenses (as defined by the SPDX license spec).

If we want to extract the information each component declared, we would need to keep that expression. Which, I think, is what the current implementation does.